### PR TITLE
Cpp polish 5

### PR DESCRIPTION
**Pull request by jtlap**
geommean has been revisited and completed

### CODE REVIEW

File: test/generator.hpp [hunk 1]
```diff
@@ -88,24 +89,24 @@ namespace eve::test
   template<typename V> auto ramp(V v)
   {
     return  [=]<typename T>(eve::as_<T>, auto&)
-            {
-              std::array<eve::element_type_t<T>, amount<T>()> d;
-              for(std::size_t i = 0;i<amount<T>();++i) d[i] = v+i;
-              return d;
-            };
+    {
```
Comment (jfalcou) [inline test/generator.hpp:92]: This is uneeded changes

File: test/generator.hpp [hunk 2]
```diff
@@ -88,24 +89,24 @@ namespace eve::test
...
     return  [=]<typename T>(eve::as_<T>, auto&)
-            {
-              using elt_t = element_type_t<T>;
-              auto n = amount<T>()-1;
-              elt_t a = n ? elt_t((v2-v1))/n : elt_t(0);
-              std::array<elt_t, amount<T>()> d;
-              for(std::size_t i = 0;i<amount<T>();++i) d[i] = a*i+v1;
-              return d;
-            };
+    {
```
Comment (jfalcou) [inline test/generator.hpp:102]: This is uneeded changes

File: test/generator.hpp [hunk 3]
```diff
@@ -114,11 +115,11 @@ namespace eve::test
   template<typename V, typename K> auto logicals(V v, K k)
   {
     return  [=]<typename T>(eve::as_<T>, auto&)
-            {
-              std::array<eve::logical<eve::element_type_t<T>>,amount<T>()> d;
-              for(std::size_t i = 0;i<amount<T>();++i) d[i] = ((v+i)%k) == 0;
-              return d;
-            };
+    {
```
Comment (jfalcou) [inline test/generator.hpp:118]: This is uneeded changes

File: test/generator.hpp [hunk 4]
```diff
@@ -166,13 +167,15 @@ namespace eve::test
...
           {
             using type  = T;
-            type nan       = eve::nan     (eve::as_<type>{});
-            type inf       = eve::inf     (eve::as_<type>{});
-            type minf      = eve::minf    (eve::as_<type>{});
-            type mzero     = eve::mzero   (eve::as_<type>{});
-            type maxflint  = eve::maxflint(eve::as_<type>{});
-            type valmax    = eve::valmax(eve::as_<type>{});
-            type valmin    = eve::valmin(eve::as_<type>{});
+            type nan         = eve::nan     (eve::as_<type>{});
```
Comment (jfalcou) [inline test/generator.hpp:170]: This is uneeded changes

File: test/generator.hpp [hunk 5]
```diff
@@ -268,7 +271,7 @@ inline bool const TTS_CAT(register_,TTS_FUNCTION) =  ::eve::test::test_setup{
     }( TYPES );                                                                                     \
                                                                                                     \
     return true;                                                                                    \
-  }} + []                                                                                           \
+   }} + []                                                                                          \
```
Comment (jfalcou) [inline test/generator.hpp:274]: This is uneeded changes

File: test/unit/module/real/core/geommean.cpp [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+#include <eve/constant/half.hpp>
+#include <eve/function/geommean.hpp>
+#include <eve/function/diff/geommean.hpp>
+#include <eve/function/sqrt.hpp>
+#include <eve/function/rsqrt.hpp>
+#include <eve/function/sign.hpp>
+#include <eve/function/sqr.hpp>
+#include <cmath>
+
+//==================================================================================================
```
Comment (jfalcou) [inline test/unit/module/real/core/geommean.cpp:20]: Remove  commented code

**Review by jfalcou — CHANGES_REQUESTED** [review]

File: test/generator.hpp (hunk 4 above)
Comment (jtlap) [reply test/generator.hpp:170]: I added zero and mindenormal

File: test/generator.hpp (hunk 1 above)
Comment (jtlap) [reply test/generator.hpp:92]: this is uneedded comment: clang-format will decide, it's just how my editor indented the text
it's perhaps unneeded but harmless
I made real changes in the file elsewhere

File: test/generator.hpp (hunk 2 above)
Comment (jtlap) [reply test/generator.hpp:102]: same answer

File: test/generator.hpp (hunk 3 above)
Comment (jtlap) [reply test/generator.hpp:118]: same answer

File: test/generator.hpp (hunk 5 above)
Comment (jtlap) [reply test/generator.hpp:274]: same answer

File: test/unit/module/real/core/geommean.cpp (hunk 6 above)
Comment (jtlap) [reply test/unit/module/real/core/geommean.cpp:20]: done

File: test/generator.hpp (hunk 1 above)
Comment (jfalcou) [reply test/generator.hpp:92]: That's the issue, your change is 1 line but the main thing I see when reviewing is blocks of tens of lines of change. Please check your local editor setup. I know we'll clans-format but this is not the reason to introduce noises that akes reviews more complex.

**Pull request merged**